### ISSUE TITLE
Support Direct placement of BlockMocks in Mock worlds.

### ISSUE THREAD

**Issue by XDMAN500** [MockBukkit/MockBukkit#1173]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the feature

As part of a unit test, I'd like to test against a block with mocked drops. The current [BlockMock](https://github.com/MockBukkit/MockBukkit/blob/219186b03fd1b2410627c9caf06423d3cdeafa6f/src/main/java/be/seeseemelk/mockbukkit/block/BlockMock.java#L580-L591) class does not have an implementation for `Block#getDrops()` and there is no way to add a specialized implementation of the BlockMock to a MockWorld.

The interface for this functionality can be an extension of the existing  [WorldMock#createBlock](https://github.com/MockBukkit/MockBukkit/blob/v1.21/src/main/java/be/seeseemelk/mockbukkit/WorldMock.java#L519C28-L519C62) method with another parameter for the BlockMock instance.

`WorldMock#createBlock(Coordinate, BlockMock)`

**Comment by Thorinwasher** [comment]
It's not better to just add something like this to the BlockMock class?:
```java
public void setBlockDrops(@NotNull Collection<ItemStack> blockDrops) {
    this.blockDrops = blockDrops.deepCopy();
}

@Override
public Collection<ItemStack> getDrops() {
    return this.blockDrops.deepCopy();
}
```

**Comment by XDMAN500** [comment]
That solves the immediate issue I'm having. Since this is a mocking framework, I don't see why the state of the WorldMock needs to be protected. Adding the createBlockmethod would still add flexibility.

**Comment by Thorinwasher** [comment]
> Since this is a mocking framework, I don't see why the state of the WorldMock needs to be protected. Adding the createBlockmethod would still add flexibility.

I don't think I understand what you mean exactly here. Do you mean that it should be possible to completely create new block materials which has custom block drops and so on, or some alternative global modification?

**Comment by XDMAN500** [comment]
No, not a global modification to all blocks.  As a MockBukkit users I want to create a test case with a block that emulates the behavior of a vanilla block.  I should be able to define the behavior of that mock block for that position in the world to write my tests.  Defining custom block mock and placing it directly into the world mock is one way to solve this problem.
